packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelConfigurationController.java

```diff
@@ -201,10 +207,12 @@ public ResponseEntity<Model> getConfiguredModel(
 			if (model.isEmpty()) {
 				throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("model.not-found"));
 			}
+			updateModelInitials(
+					model.get().getInitials(), modelConfiguration.get().getInitialSemanticList());
+			updateModelParameters(
+					model.get().getParameters(), modelConfiguration.get().getParameterSemanticList());
```

Comment:
Strictly speaking, you want to do in order of
- parameters
- initials
- observables

Because initials depends on parameters and observables depends on initials and parameters. Also you seem to be missing observables.

Reply:
Sure Ill go in that order.  We weren't using observables in the model config drilldown on the front end so far which is why I hadn't created it, but I can add that in.